pkg/config/expand.go

```diff
@@ -513,20 +512,46 @@ type varContext struct {
 	blueprint  Blueprint
 }
 
+type reference interface {
+	validate(Blueprint) error
+	isIntergroup() bool
+	String() string
+	getFromModuleID() string
+	getToModuleID() string
+}
+
 /*
 A module reference is made by the use keyword and is subject to IGC constraints
 of references (ordering, explicitness). It has the following fields:
-  - ID: a module ID
+  - ToModuleID: the target module ID
+  - FromModuleID: the source module ID
   - ToGroupID: the deployment group in which the module is *expected* to be found
   - FromGroupID: the deployment group from which the reference is made
   - Explicit: a boolean value indicating whether the user made a reference that
     explicitly identified ToGroupID rather than inferring it using FromGroupID
 */
 type modReference struct {
-	ID          string
-	ToGroupID   string
-	FromGroupID string
-	Explicit    bool
+	ToModuleID   string
+	FromModuleID string
+	ToGroupID    string
+	FromGroupID  string
+	Explicit     bool
+}
+
+func (ref modReference) String() string {
+	return ref.ToGroupID + "." + ref.ToModuleID
+}
+
+func (ref modReference) isIntergroup() bool {
+	return ref.ToGroupID == ref.FromGroupID
+}
+
+func (ref modReference) getFromModuleID() string {
+	return ref.FromModuleID
+}
+
+func (ref modReference) getToModuleID() string {
+	return ref.FromModuleID
```

Comment:
I believe this is returning the wrong variable.

Reply:
Modified on PR branch.

Reply:
I believe this is done, close if you agree.